lib/counter_culture/reconciler.rb

```diff
@@ -252,7 +252,11 @@ def join_clauses(where)
           if index == reverse_relation.size - 1
             # conditions must be applied to the join on which we are counting
             if where
-              joins_sql += " AND (#{model.send(:sanitize_sql_for_conditions, where)})"
+              if where.is_a? Symbol
+                joins_sql += " AND #{target_table_alias}.#{model.primary_key} IN (#{model.send(where).pluck(model.primary_key).join(',')})"
```

Comment:
I think that would simplify your work here, too, because you can get the SQL from the scope?

Reply:
I updated the code to this.

```
if where.respond_to?(:to_sql)
  joins_sql += " AND #{target_table_alias}.#{model.primary_key} IN (#{where.select(model.primary_key).to_sql})"
else
  joins_sql += " AND (#{model.send(:sanitize_sql_for_conditions, where)})"
end
```

I have no idea about simpler way.
Scope is able to contain JOIN cause, So I have to use IN cause and subquery.

Reply:
Yeah I think this is good.